woocommerce/woocommerce-ios
# [Order form] Remove `ProductRowViewModel` from `CollapsibleProductRowCardViewModel`

Part of #11357


## Why
This is the last step to disentangle `ProductRowViewModel` from `CollapsibleProductRowCardViewModel`. After this PR, the two view models don't rely on each other's data and the order form can be updated without potential side effects in `ProductRowViewModel`.

## How
* Adds three remaining properties to `CollapsibleProductRowCardViewModel`:
   * `discount` to hold any discount added to the product in the order.
   * `id` to set a unique ID on the view model, rather than getting it from `ProductRowViewModel`. (This is still set as the order item ID, which is what we were setting for `ProductRowViewModel.id`.)
   * `productOrVariationID` to hold a reference to the ID of the product or variation added to the order. This is currently only used in unit tests.
* Changes `CollapsibleProductCardPriceSummaryViewModel` to a class instead of a struct. This allows us to set `quantity` as a published property and update it when `ProductStepperViewModel.quantity` changes (the source of truth for the quantity in the order).
* Removes the `productViewModel` property.

## Testing instructions


There should be no changes in the order form UX:

- Go to the Orders tab
- Tap on `+` to create an order
- Add at least one product to the order
- Expand any product card
- Change the quantity -> Confirm the "Price" line updates as expected
- Add a discount -> Confirm the discount is added as expected
- Tap "Create" to create the order -> Confirm the created order contains the products and quantities you added


---
